src/codegen/query_compiler.cpp

```diff
@@ -88,6 +90,11 @@ bool QueryCompiler::IsSupported(const planner::AbstractPlan &plan) {
       pred = scan_plan.GetPredicate();
       break;
     }
+    case PlanNodeType::INDEXSCAN: {
```

Comment:
Merge this case with SeqScanPlan, cast to `AbstractScanPlan` and check predicate.